ecosystem/indexer-grpc/indexer-grpc-fullnode/src/stream_coordinator.rs

```diff
@@ -0,0 +1,384 @@
+                    // continue;
+                    panic!(
+                        "Could not convert txn {} from OnChainTransactions: {:?}",
+                        txn_version, err
+                    );
+                },
+            }
+        }
+
+        if transactions.is_empty() {
+            panic!("No transactions!");
+        }
+
+        let fetch_millis = (chrono::Utc::now().naive_utc() - start_millis).num_milliseconds();
+
+        info!(
+            first_version = first_version,
+            num_transactions = transactions.len(),
+            time_millis = fetch_millis,
+            actual_last_version = transactions
+                .last()
+                .map(|txn| txn.version().unwrap())
+                .unwrap_or(0),
+            "Fetched transactions",
+        );
+
+        FETCHED_TRANSACTION.inc();
+        transactions
+    }
+
+    fn convert_to_pb_txns(api_txns: Vec<APITransaction>) -> Vec<TransactionPB> {
+        api_txns
+            .iter()
+            .map(|txn| {
+                let info = txn.transaction_info().unwrap();
+                convert_transaction(txn, info.block_height.unwrap().0, info.epoch.unwrap().0)
+            })
+            .collect()
+    }
+
+    fn encode_pb_txns(pb_txns: Vec<TransactionPB>) -> Vec<TransactionOutput> {
+        pb_txns
+            .iter()
+            .map(|txn| {
+                let mut buf = vec![];
+                txn.encode(&mut buf).unwrap_or_else(|_| {
+                    panic!(
+                        "Could not convert protobuf transaction to bytes '{:?}'",
+                        txn
+                    )
+                });
+                let encoded_proto_data = base64::encode(buf);
+                TransactionOutput {
+                    encoded_proto_data,
+                    version: txn.version,
+                    timestamp: txn.timestamp.clone(),
+                }
+            })
+            .collect()
+    }
+
+    pub fn set_highest_known_version(&mut self) -> anyhow::Result<()> {
+        let info = self.context.get_latest_ledger_info_wrapped()?;
+        self.highest_known_version = info.ledger_version.0;
+        Ok(())
+    }
+
+    /// Will keep looping and checking the latest ledger info to see if there are new transactions
+    /// If there are, it will set the highest known version
+    async fn ensure_highest_known_version(&mut self) {
+        let mut empty_loops = 0;
+        while self.highest_known_version == 0 || self.current_version > self.highest_known_version {
+            if empty_loops > 0 {
+                tokio::time::sleep(Duration::from_millis(RETRY_TIME_MILLIS)).await;
+            }
+            empty_loops += 1;
+            if let Err(err) = self.set_highest_known_version() {
+                error!(
+                    error = format!("{:?}", err),
+                    "Failed to set highest known version"
+                );
+                continue;
```

Comment:
useless continue

Reply:
lol yea legacy will remove future PR so @larry-aptos doesn't have to rebase.